Error when compiling the DelaunayMeshingApplication

**Description**
Here the error:

```
/usr/bin/ld: cannot find -ltriangle
collect2: error: ld returned 1 exit status
applications/DelaunayMeshingApplication/CMakeFiles/KratosDelaunayMeshingCore.dir/build.make:281: recipe for target 'applications/DelaunayMeshingApplication/libKratosDelaunayMeshingCore.so' failed
make[2]: *** [applications/DelaunayMeshingApplication/libKratosDelaunayMeshingCore.so] Error 1
CMakeFiles/Makefile2:3413: recipe for target 'applications/DelaunayMeshingApplication/CMakeFiles/KratosDelaunayMeshingCore.dir/all' failed
make[1]: *** [applications/DelaunayMeshingApplication/CMakeFiles/KratosDelaunayMeshingCore.dir/all] Error 2
```

**Scope**
Where in Kratos is this bug happening. 
E.g.
- Kratos DelaunayMeshingApplication
- As a dependency to PfemFluidDynamicsApplication

**Environment**
Please specify your operating system, kratos branch, python version or any other library versions that you consider relevant.
E.g.:
- OS: Ubuntu 18.04
- Branch: master

--- a/applications/DelaunayMeshingApplication/CMakeLists.txt
+++ b/applications/DelaunayMeshingApplication/CMakeLists.txt
@@ -15,7 +15,7 @@ if(NOT DEFINED ${INCLUDE_TRIANGLE})
     set(INCLUDE_TRIANGLE ON)
     set(TRIANGLE_DIR "${KRATOS_SOURCE_DIR}/external_libraries/triangle")
   else(${USE_TRIANGLE_NONFREE_TPL} MATCHES ON )
-    message(STATUS "INCLUDE_TRIANGLE not defined, neither USE_TRIANGLE_NONFREE_TPL=ON is defined. The application DelaunayMeshingApplication will not compile")
+    message(FATAL_ERROR "INCLUDE_TRIANGLE not defined, neither USE_TRIANGLE_NONFREE_TPL=ON is defined. The application DelaunayMeshingApplication will not compile")
   endif(${USE_TRIANGLE_NONFREE_TPL} MATCHES ON )
 endif(NOT DEFINED ${INCLUDE_TRIANGLE})
 

--- a/applications/DelaunayMeshingApplication/CMakeLists.txt
+++ b/applications/DelaunayMeshingApplication/CMakeLists.txt
@@ -10,8 +10,13 @@ include(pybind11Tools)
 ## Set a default value for triangle and tetgen in case the user does not set it
 ## to avoid problems with the define linkage block. By default we will compile the lib
 if(NOT DEFINED ${INCLUDE_TRIANGLE})
-  set(INCLUDE_TRIANGLE ON)
-  set(TRIANGLE_DIR "${KRATOS_SOURCE_DIR}/external_libraries/triangle")
+  # Compiling the triangle library
+  if(${USE_TRIANGLE_NONFREE_TPL} MATCHES ON )
+    set(INCLUDE_TRIANGLE ON)
+    set(TRIANGLE_DIR "${KRATOS_SOURCE_DIR}/external_libraries/triangle")
+  else(${USE_TRIANGLE_NONFREE_TPL} MATCHES ON )
+    message(FATAL_ERROR "INCLUDE_TRIANGLE not defined, neither USE_TRIANGLE_NONFREE_TPL=ON")
+  endif(${USE_TRIANGLE_NONFREE_TPL} MATCHES ON )
 endif(NOT DEFINED ${INCLUDE_TRIANGLE})
 
 if(NOT DEFINED ${INCLUDE_TETGEN})
@@ -24,7 +29,6 @@ if(${INCLUDE_TRIANGLE} MATCHES ON)
   include_directories(${TRIANGLE_DIR})
   list(APPEND MESHER_LIBRARIES triangle)
 endif(${INCLUDE_TRIANGLE} MATCHES ON)
-
 
 if(${INCLUDE_TETGEN} MATCHES ON)
   set(EXTERNAL_LIBRARIES_DIR "${CMAKE_CURRENT_SOURCE_DIR}/external_libraries")

--- a/applications/DelaunayMeshingApplication/CMakeLists.txt
+++ b/applications/DelaunayMeshingApplication/CMakeLists.txt
@@ -15,7 +15,7 @@ if(NOT DEFINED ${INCLUDE_TRIANGLE})
     set(INCLUDE_TRIANGLE ON)
     set(TRIANGLE_DIR "${KRATOS_SOURCE_DIR}/external_libraries/triangle")
   else(${USE_TRIANGLE_NONFREE_TPL} MATCHES ON )
-    message(FATAL_ERROR "INCLUDE_TRIANGLE not defined, neither USE_TRIANGLE_NONFREE_TPL=ON")
+    message(STATUS "INCLUDE_TRIANGLE not defined, neither USE_TRIANGLE_NONFREE_TPL=ON is defined. The application DelaunayMeshingApplication will not compile")
   endif(${USE_TRIANGLE_NONFREE_TPL} MATCHES ON )
 endif(NOT DEFINED ${INCLUDE_TRIANGLE})
 

